### DISCUSSION TITLE
Is there a way to trees shake exports from a file from the context of a chunk

### DISCUSSION THREAD

**Discussion by AmalJossy** [webpack/webpack#16101] [Q&A] [answered]
A react project at my org has a pattern where a single file does a lot of re-exporting. What I noticed is that all the exports and their dependencies go into the initial chunk because it uses a few of these exports. Most of the re-exports are actually used by dynamically loaded modules and are not needed in most cases but still are loaded every time because they are part of the entry chunk.
I have created a [sandbox to demo](https://codesandbox.io/s/dependencies-escaping-async-chunk-tu12ey?file=/src/utils.js).

Here the blueMethod is only needed for the async chunk but is still added to the initial chunk. 

Is there a way we can optimize such cases without refactoring the file into smaller files? may be by eliminating the exports not used by the particular chunk?

**Comment by vankop** [comment, marked as answer]
> Here the blueMethod is only needed for the async chunk but is still added to the initial chunk.
Is there a way we can optimize such cases without refactoring the file into smaller files? may be by eliminating the exports not used by the particular chunk?

no, webpack can't optimize on code level (e.g. one exports in one chunk, other exports in another chunk), only on module level.. 

Also this optimization not always possible + this will require changes in how webpack bootstrap modules..
